### PR TITLE
homework 5

### PR DESCRIPTION
**Pull request by DSVAG**

### CODE REVIEW

File: app/src/main/java/com/dsvag/androidacademyproject/ui/fragments/MovieListFragment.kt [hunk 1]
```diff
@@ -4,78 +4,43 @@ import android.os.Bundle
...
     private val movieAdapter by lazy { MovieAdapter() }
+
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?
     ): View {
         _binding = FragmentMovieListBinding.inflate(inflater, container, false)
 
-        movieAdapter.setData(generateMovies())
-
         binding.moveList.setHasFixedSize(true)
```
Comment (pcfaktor) [inline app/src/main/java/com/dsvag/androidacademyproject/ui/fragments/MovieListFragment.kt:29]: Maybe increase spanCount for landscape mode?

**Review by pcfaktor — APPROVED** [review]
1. Split the "fragments" package by screens;
2. Move each adapter to its screen package (adapter it isn't data layer abstraction);
3. Also, move ViewBinding.kt into the UI layer.
4. Actually, models also aren't abstractions of the data layer.

File: app/src/main/java/com/dsvag/androidacademyproject/data/di/AppComponent.kt [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package com.dsvag.androidacademyproject.data.di
+
+import android.app.Application
+import com.dsvag.androidacademyproject.data.repositories.MovieRepository
+
```
Comment (pcfaktor) [inline app/src/main/java/com/dsvag/androidacademyproject/data/di/AppComponent.kt:5]: Maybe it's time to use DI Framework?

File: app/src/main/java/com/dsvag/androidacademyproject/data/di/App.kt [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package com.dsvag.androidacademyproject.data.di
+
+import android.app.Application
+import android.content.Context
+
```
Comment (pcfaktor) [inline app/src/main/java/com/dsvag/androidacademyproject/data/di/App.kt:5]: Why the App in the data layer? The root of the source would be a better place for it.

File: app/src/main/java/com/dsvag/androidacademyproject/ui/viewmodels/MoviesViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,31 @@
+package com.dsvag.androidacademyproject.ui.viewmodels
+
+import android.app.Application
+import androidx.lifecycle.*
+import com.dsvag.androidacademyproject.data.di.getAppComponent
+import com.dsvag.androidacademyproject.data.repositories.MovieRepository
+import com.dsvag.androidacademyproject.models.Movie
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
```
Comment (pcfaktor) [inline app/src/main/java/com/dsvag/androidacademyproject/ui/viewmodels/MoviesViewModel.kt:10]: One of the main points of ViewModel is that your data can survive the configuration changes in it. But you only get data in it.

**Review by pcfaktor — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/dsvag/androidacademyproject/data/di/AppComponent.kt (hunk 2 above)
Comment (DSVAG) [reply app/src/main/java/com/dsvag/androidacademyproject/data/di/AppComponent.kt:5]: Hilt or dagger?

**Review by pcfaktor — APPROVED** [review]
LGTM

**Pull request merged**